activemq-broker/src/main/java/org/apache/activemq/security/jwt/JwtAuthenticationPlugin.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.security.jwt;
+
+import com.nimbusds.jose.JWSAlgorithm;
+import org.apache.activemq.broker.Broker;
+import org.apache.activemq.broker.BrokerPlugin;
+
+/**
+ * A simple JWT plugin giving ActiveMQ the ability to support JWT tokens to authenticate and authorize users
+ *
+ * @org.apache.xbean.XBean element="jwtAuthenticationPlugin"
+ *                         description="Provides a JWT authentication plugin"
+ *
+ *
+ * This plugin is rather simple and is only meant to be used as a first step. In real world applications, we would need
+ * to being able to specify different key format (RSA, DSA, EC, etc), the issuer, the claim to use for groups and maybe
+ * some mapping functionalities.
+ * The header name is also hard coded for simplicity.
+ */
+public class JwtAuthenticationPlugin implements BrokerPlugin {
```

Comment:
Perhaps I'm wrong, but this doesn't appear usable as is. How can users configure these values? Is the idea here that they will need implement and deploy their own version of this class in order to configure auth as needed? If so, are there plans to make this configurable in the future?